subxt/src/tx/tx_client.rs

```diff
@@ -444,12 +448,12 @@ where
     /// Returns a [`TxProgress`], which can be used to track the status of the transaction
     /// and obtain details about it, once it has made it into a block.
     pub async fn submit_and_watch(&self) -> Result<TxProgress<T, C>, Error> {
-        // Get a hash of the extrinsic (we'll need this later).
-        let ext_hash = T::Hasher::hash_of(&self.encoded);
-
+        // Get a hash of the extrinsic (we'll need this later)
+        // We use the `encoded_prefixed` to calculate the hash, because it has the same hash
```

Comment:
Here, I think the hash should be as it was before; ie the hash of the exact bytes we're sending to the `watch_extrinsic` rpc call and not a hash of those bytes with another length prefixed.

To check, the hash returned here should be identical to the hash returned from making the `submit` call below with the same extrinsic; worth trying that to make sure one way or the other :)